docs/rfcs/0004-ecs-deployment.md

```diff
@@ -0,0 +1,82 @@
+
+With `EXTERNAL` deployment controller is used and required by PipeCD workflow, the above simplest case will be changed to:
+- Container's image
+- Task Definition which registered with AWS ECS service
+- Service which has to define 3 most important spec: `DesiredCount`, `SchedulingStrategy` and `DeploymentController` which always be set to `EXTERNAL`. The `service.TaskDefinitionArn` should be set to null and it's unable to changed in case deployment controller of type `EXTERNAL` is used.
+- TaskSet required to create Task with specified TaskDefinition, those Task will be controlled by Service after created (as it's replica set function). 
+
+Insprised by the [AWS Blue/Green deployments with ECS External Deployment Controller](https://aws.amazon.com/blogs/containers/blue-green-deployments-with-the-ecs-external-deployment-controller/), in case of PipeCD's AWS ECS deployments, all current common stages (`WAIT`, `WAIT_APPROVAL`, `ANALYSIS`) are inherited, besides with stages for ECS deployment such as `ECS_SYNC`, `ECS_CANARY_ROLLOUT` and `ECS_TRAFFIC_ROUTING`.
+
+- `ECS_SYNC` simply applies a service definition and a task definition.
+- `ECS_CANARY_ROLLOUT` deploys workloads of the new version.
+- `ECS_TRAFFIC_ROUTING` changes the configuration of the load balancer (you have to prepare and configure it in `.pipe.yaml` to enable using this stage) in order to change the traffic routing state.
+
+A sample `.pipe.yaml` for PipeCD ECS Application would look like:
+
+```yaml
+apiVersion: pipecd.dev/v1beta1
+kind: ECSApp
+spec:
+  input:
+    name: Sample
+    serviceDefinition: path/to/servicedef.yaml
+    taskDefinition: path/to/taskdef.yaml
+    loadBalancers:
+        - targetGroupArn: {PRIMARY_TARGET_GROUP_ARN}
+          containerName: sample-app
+          containerPort: 80
+  pipeline:
+    stages:
+      # Deploy workloads of the new version.
+      # But this is still receiving no traffic.
+      - name: ECS_CANARY_ROLLOUT
+      # Change the traffic routing state where
+      # the new version will receive the specified percentage of traffic.
+      # This is known as multi-phase canary strategy.
+      - name: ECS_TRAFFIC_ROUTING
+        with:
+          canary: 10
+      # Optional: We can also add an ANALYSIS stage to verify the new version.
+      # If this stage finds any not good metrics of the new version,
+      # a rollback process to the previous version will be executed.
+      - name: ANALYSIS
+      # Change the traffic routing state where
+      # the new version will receive 100% of the traffic.
+      - name: ECS_TRAFFIC_ROUTING
+        with:
+          canary: 100
+```
+
```

Comment:
Please add more examples for quick sync, canary, blue-green to verify the definition of stages.